syntax = "proto3";
package producer;


// Service
service kafkaProducer {
  rpc producer(stream producerRequest) returns (stream producerResponse) {}
  rpc getAllProducerProcesses(getAllProducerProcessesRequest) returns (getAllProducerProcessesResponse) {}
  rpc stopProducer(stopProducerRequest) returns (stopProducerResponse) {}
  rpc stopAllProducerProcesses(stopAllProducerProcessesRequest) returns (stopAllProducerProcessesResponse) {}
}


// Request Name
message producerRequest {
  string data=1;
}
message getAllProducerProcessesRequest{
  string data=1;
}
message stopProducerRequest {
  string process_name=1;
}
message stopAllProducerProcessesRequest{
  string data=1;
}


// Response Array
message producerResponse{
  string result=1;
  string process_name=2;
  bytes frame=3;
}
message getAllProducerProcessesResponse{
  string data=1;
}
message stopProducerResponse {
  string result=1;
}
message stopAllProducerProcessesResponse{
  string result=1;
}

